src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -739,30 +731,62 @@ void Lowering::LowerModPow2(GenTree* node)
     GenTree* dividend2 = comp->gtClone(dividend);
     BlockRange().InsertAfter(dividend, dividend2);
 
-    GenTreeIntCon* cns = comp->gtNewIconNode(cnsValue, type);
-    BlockRange().InsertAfter(dividend2, cns);
-
-    GenTree* const trueExpr = comp->gtNewOperNode(GT_AND, type, dividend, cns);
-    BlockRange().InsertAfter(cns, trueExpr);
-    LowerNode(trueExpr);
+    if (divisorCnsValue == 2)
+    {
+        // {expr} % 2
 
-    GenTree* const neg = comp->gtNewOperNode(GT_NEG, type, dividend2);
-    neg->gtFlags |= GTF_SET_FLAGS;
-    BlockRange().InsertAfter(trueExpr, neg);
+        GenTreeIntCon* cns = comp->gtNewIconNode(divisorCnsValueMinusOne, type);
```

Comment:
Some of the things like below can be common?

```
        GenTreeIntCon* cns = comp->gtNewIconNode(divisorCnsValueMinusOne, type);
        BlockRange().InsertAfter(dividend2, cns);

        GenTree* const trueExpr = comp->gtNewOperNode(GT_AND, type, dividend, cns);
        BlockRange().InsertAfter(cns, trueExpr);
        LowerNode(trueExpr);
```

Reply:
Yea, looks like they can be lifted up